### PR TITLE
feat: add `series.rename`

### LINKED ISSUE #1051: feat: series.rename
I keep reaching for this and realising it doesn't exist

reference: https://docs.pola.rs/api/python/stable/reference/series/api/polars.Series.rename.html

### PR DESCRIPTION
**Pull request by raisadz**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #1051

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/series.py [hunk 1]
```diff
@@ -1216,6 +1216,47 @@ def alias(self, name: str) -> Self:
...
+            >>> func(s_pl)  # doctest: +NORMALIZE_WHITESPACE
+            shape: (3,)
+            Series: 'bar' [i64]
+            [
+               1
+               2
+               3
+            ]
+        """
+        return self._from_compliant_series(self._compliant_series.alias(name=name))
```
Comment (FBruzzesi) [inline narwhals/series.py:1258]: Should just be the same of

```suggestion
        return self.alias(name=name)
```
?

**Review by FBruzzesi — COMMENTED** [review]
Thanks @raisadz 🚀

It looks good! I left a comment to simplify a bit the logic. 
I think `./utils/generate_backend_completeness.py` may need some love to render that `rename` is implemented for all series. But I would be ok to leave that as a follow up as well 👌

**Comment by raisadz** [comment]
> Thanks @raisadz 🚀
> 
> It looks good! I left a comment to simplify a bit the logic. I think `./utils/generate_backend_completeness.py` may need some love to render that `rename` is implemented for all series. But I would be ok to leave that as a follow up as well 👌

Thank you for your review @FBruzzesi !

File: narwhals/series.py [hunk 2]
```diff
@@ -1216,6 +1216,47 @@ def alias(self, name: str) -> Self:
...
+            >>> s_pd = pd.Series(s, name="foo")
+            >>> s_pl = pl.Series("foo", s)
+
+            We define a library agnostic function:
+
+            >>> @nw.narwhalify
+            ... def func(s):
+            ...     return s.rename("bar")
+
+            We can then pass either pandas or Polars to `func`:
```
Comment (FBruzzesi) [inline narwhals/series.py:1242]: Hey @raisadz thanks for adjusting!

I have one more tiny request: in #1000 we are trying to add pyarrow to the docstrings. Would you mind adding an example with pyarrow as well? Maybe we don't have an example with series yet, but you can create a pyarrow "series" via `s_pa = pa.chunked_array([s])`.

Then the line here would become:

```suggestion
            We can then pass any supported library such as pandas, Polars, or PyArrow:
```
and finally the rendering below 🙃

Comment (raisadz) [reply narwhals/series.py]: Thanks for the suggestion! I added an example with PyArrow series to the docstrings

**Review by FBruzzesi — APPROVED** [review]
Awesome, thanks @raisadz ! Let's ship it!

CatFunnycatGIF

**Pull request merged**